src/gretel_synthetics/generate.py

```diff
@@ -7,106 +7,32 @@
     - Trained a model
 """
 import logging
-from collections import namedtuple
-from dataclasses import dataclass, asdict
-from typing import Tuple, TYPE_CHECKING, List, Callable
+from typing import TYPE_CHECKING, Callable
 
-import sentencepiece as spm
 import tensorflow as tf
 
-from gretel_synthetics.model import _build_sequential_model
+from . import generate_parallel, generator
 
 if TYPE_CHECKING:  # pragma: no cover
-    from gretel_synthetics.config import BaseConfig
+    from gretel_synthetics.config import LocalConfig
```

Comment:
This was done before you worked on this PR, but we recently just discovered setting the imports to `None` if not doing Type checking which means we can still use the obj reference for type hinting later. Do you mind making that change here too?